### DISCUSSION TITLE
Giskard in Unsupervised Settings

### DISCUSSION THREAD

**Discussion by cpinon-grd** [Giskard-AI/giskard-oss#1593] [Q&A] [answered]
Hello, 

I am checking this amazing tool, and I was wondering if it is possible to integrate it in unsupervised learning settings (I'm thinking of anomaly detection systems), where labels are not available, in order to detect possible problems. 

Thank you in advance!

**Comment by luca-martial** [comment, marked as answer]
Hi @cpinon-grd, thanks for reaching out here! Unfortunately we don't yet handle unsupervised learning models. This will require quite a bit of customization at the moment, but we're always interested in hearing your thoughts on the types of vulnerabilities you would be curious to look into for such systems (anomaly detection). What types of issues have you experienced?

**Reply by cpinon-grd** [reply]
Hey, thanks for replying!

I am currently starting some research related to making unsupervised anomaly detection (applied to cybersecurity) more robust, auditable and debuggable and I came across Giskard. Some of the gaps/issues in the field I've identified at the moment are:

- Difficulty in defining normal behavior
- Imbalanced data
- Noise in the data
- Dynamic nature of anomalies
- Evaluation and validation challenges due to the lack of labels
- Threshold determination difficulties


I don't have many specific use cases or necessities yet, but I'll let you know if something comes to my mind related to the functionalities of Giskard!

**Reply by luca-martial** [reply]
That's really insightful, thanks a lot @cpinon-grd ! Feel free to join our Discord community: https://docs.giskard.ai/en/latest/community/discord/index.html

We can stay in touch on there and co-build for unsupervised learning 🤝
